core/sail/lmdb/src/main/java/org/eclipse/rdf4j/sail/lmdb/TripleStore.java

```diff
@@ -668,10 +668,9 @@ public boolean storeTriple(long subj, long pred, long obj, long context, boolean
 	 * @return A mapping of each modified context to the number of statements removed in that context.
 	 * @throws IOException
 	 */
-	public Map<Long, Long> removeTriplesByContext(Txn txn, long subj, long pred, long obj, long context,
-			boolean explicit)
-			throws IOException {
-		RecordIterator records = getTriples(txn, subj, pred, obj, context, explicit);
+	public Map<Long, Long> removeTriplesByContext(long subj, long pred, long obj, long context,
+			boolean explicit) throws IOException {
+		RecordIterator records = getTriples(txnManager.createTxn(writeTxn), subj, pred, obj, context, explicit);
```

Comment:
Does this transaction need to be closed?

Reply:
No, its just a wrapper object for the (managed) writeTxn.